internal/cluster/distribution/eks/eksprovider/workflow/activity_calculate_node_pool_version.go

```diff
@@ -62,11 +64,59 @@ func (a CalculateNodePoolVersionActivity) Execute(
 	}
 
 	return CalculateNodePoolVersionActivityOutput{
-		Version: eks.CalculateNodePoolVersion(
+		Version: calculateNodePoolVersion(
 			input.Image,
 			volumeEncryption,
 			fmt.Sprintf("%d", input.VolumeSize),
 			strings.Join(input.CustomSecurityGroups, ","),
 		),
 	}, nil
 }
+
+// CalculateNodePoolVersion retrieves the calculated nodePoolVersion
+//
+// This is a convenience wrapper around the corresponding activity.
+func CalculateNodePoolVersion(
+	ctx workflow.Context,
+	image string,
+	volumeEncryption *eks.NodePoolVolumeEncryption,
+	volumeSize int,
+	customSecurityGroups []string,
+) (string, error) {
+	var activityOutput CalculateNodePoolVersionActivityOutput
+	err := calculateNodePoolVersionAsync(
+		ctx, image, volumeEncryption, volumeSize, customSecurityGroups).Get(ctx, &activityOutput)
+	if err != nil {
+		return "", err
+	}
+
+	return activityOutput.Version, nil
+}
+
+// calculateNodePoolVersion retrieves a future object for calucating the  nodePoolVersion
+//
+// This is a convenience wrapper around the corresponding activity.
+func calculateNodePoolVersionAsync(
+	ctx workflow.Context,
+	image string,
+	volumeEncryption *eks.NodePoolVolumeEncryption,
+	volumeSize int,
+	customSecurityGroups []string,
+) workflow.Future {
+	return workflow.ExecuteActivity(ctx, CalculateNodePoolVersionActivityName, CalculateNodePoolVersionActivityInput{
+		Image:                image,
+		VolumeEncryption:     volumeEncryption,
+		VolumeSize:           volumeSize,
+		CustomSecurityGroups: customSecurityGroups,
+	})
+}
+
+func calculateNodePoolVersion(input ...string) string {
```

Comment:
**Recommended**: as we want to avoid letting other activities even from this package use the function directly, I suggest to inline the functionality of this function into the activity execution, because otherwise it's going to name clash with the unexported wrappers